File: src/runtime/api-keys.ts
import type { AuthenticatedPrincipal } from '../contract/principals.js';
import type { ApiKeyService } from '../api-keys/service.js';

export async function authenticateBearerToken(
  authorizationHeader: string,
  service: ApiKeyService,
): Promise<AuthenticatedPrincipal | null> {
  const [scheme, token] = authorizationHeader.split(/\s+/, 2);
  if (!scheme || !token || scheme.toLowerCase() !== 'bearer') {
    return null;
  }
  return service.authenticateApiKey(token);
}

File: src/appport_services/runtime/api_keys.py
from __future__ import annotations

from appport_services.api_keys.service import ApiKeyService
from appport_services.contract.principals import AuthenticatedPrincipal


def authenticate_bearer_token(
    authorization_header: str,
    service: ApiKeyService,
) -> AuthenticatedPrincipal | None:
    scheme, _, token = authorization_header.partition(" ")
    if scheme.lower() != "bearer" or not token:
        return None
    return service.authenticate_api_key(token)
